fees/eastern-neighborhoods-non-residential-open-space/viewmodel.js

```diff
@@ -0,0 +1,76 @@
+define([
+    'knockout',
+    'fees/abstract-fee',
+    'json!./settings.json',
+    './component'
+], function(ko, AbstractFee, settings) {
+    var EasternNeighborhoodsNonResidentialOpenSpaceFee = function(params) {
+        this.settings = settings;
+
+        AbstractFee.apply(this, [params]);
+
+        this.triggered = ko.computed(function() {
+            return this.isProjectInArea() &&
+                (
+                    this.newNonRes() > 0 ||
+                    this.newNonRes()/this.existingNonResGFA() >= .2 ||
+                    this.nonResGFA()/this.existingNonResGFA() >= .2 ||
+                    this.officeGFA()/this.existingNonResGFA() >= .2
+
+                );
+        }, this);
+
+        this.ready = ko.computed(function() {
+            if (!this.triggered()) {
+                return true;
+            }
+            return this.retailGFA() !== null && this.retailGFA() !== '' &&
+                this.retailOpenSpaceProvidedGFA() !== null && this.retailOpenSpaceProvidedGFA() !== '' &&
+                this.manufacturingGFA() !== null && this.manufacturingGFA() !== '' &&
+                this.manufacturingOpenSpaceProvidedGFA() !== null && this.manufacturingOpenSpaceProvidedGFA() !== '' &&
+                this.officeGFA() !== null && this.officeGFA() !== '' &&
+                this.officeOpenSpaceProvidedGFA() !== null && this.officeOpenSpaceProvidedGFA() !== '' &&
+                this.otherGFA() !== null && this.otherGFA() !== '' &&
+                this.otherOpenSpaceProvidedGFA() !== null && this.otherOpenSpaceProvidedGFA() !== '';
+        }, this);
+
+
+        this.calculatedFee = ko.computed(function() {
+            if (!this.triggered()) {
+                return 0;
+            }
+
+            this.calculateRequiredOpenSpaceOrFee = function(a, b, c) {
```

Comment:
@ryan86 what are `a`, `b`, and `c` here? better naming would be helpful for comprehension.  Also, using eval is always discouraged, and almost never necessary.  Why are you using eval here? (it's hard to understand the intention behind this function currently).  Finally, you should not be creating/overriding properties of your fee inside of this computed function, as they will be overridden every time the computed is evaluated and their creation is not actually part of the calculation.